src/login/logind-session.c

```diff
@@ -1147,6 +1170,8 @@ static void session_restore_vt(Session *s) {
         (void) ioctl(vt, VT_SETMODE, &mode);
         (void) fchown(vt, 0, (gid_t) -1);
 
+        assert_se(release_terminal_fd(vt) >= 0);
```

Comment:
what's the reason for releasing it only now? i.e. isn't it enough to give it up right-away again?

Reply:
That should help ensuring that no other processes become the controlling process while we're restoring the VT. Otherwise the race wouldn't be fixed.